wls-gui-wahllokalsystem/tests/composables/wahl/wahlService.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { useCommonTestDataFactory } from "@tests/utils/common/CommonTestDataFactory.ts";
+import { useWahlTestDataFactory } from "@tests/utils/wahl/WahlTestDataFactory.ts";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+
+import { useWahlService } from "@/composables/wahl/wahlService.ts";
+
+const { createWahl, createWahlDTO } = useWahlTestDataFactory();
+
+const mockDefinitions = vi.hoisted(() => ({
+  getWahlen: vi.fn(),
+  addNotification: vi.fn(),
+  mapDtoToModel: vi.fn(),
+  wahlDTOWahlartEnum: vi.fn(),
+  configurationConstructor: vi.fn().mockImplementation(() => ({})),
+}));
+
+vi.mock("@/api/wls-clients/generated-basisdaten-api", () => ({
+  WahlenControllerApi: vi.fn().mockImplementation(() => ({
+    getWahlen: mockDefinitions.getWahlen,
+  })),
+  Configuration: mockDefinitions.configurationConstructor,
+  WahlDTOWahlartEnum: mockDefinitions.wahlDTOWahlartEnum,
+}));
+vi.mock("@/composables/wahl/wahlMapper.ts", () => ({
+  useWahlMapper: () => ({
+    toModel: mockDefinitions.mapDtoToModel,
+  }),
+}));
+
+vi.mock("@/composables/userNotification/userNotificationService.ts", () => ({
+  useUserNotificationService: () => ({
+    addNotification: mockDefinitions.addNotification,
+  }),
+}));
+
+const { generateRandomString } = useCommonTestDataFactory();
+
+describe("WahlService.ts", () => {
+  const { getWahlen } = useWahlService();
+
+  beforeEach(() => {
+    vi.resetAllMocks();
+    vi.clearAllMocks();
+  });
+
+  it("should_returnWahl_when_wahltagIDIsGiven", async () => {
+    const wahltagID = generateRandomString(10);
+    mockDefinitions.getWahlen.mockReturnValue(
+      Promise.resolve({ status: 200, data: [createWahlDTO()] })
+    );
+    const mockedMappedWahl = createWahl();
+    mockDefinitions.mapDtoToModel.mockReturnValue(mockedMappedWahl);
+
+    const result = await getWahlen(wahltagID);
+
+    expect(result).toEqual([mockedMappedWahl]);
+    expect(mockDefinitions.getWahlen.mock.calls.length).toStrictEqual(1);
+    expect(mockDefinitions.getWahlen.mock.calls).toStrictEqual([[wahltagID]]);
```

Comment:
die untere Assertion umfasst die Prüfung auf length da dein Array die Länge 1 hat.
```suggestion
    expect(mockDefinitions.getWahlen.mock.calls).toStrictEqual([[wahltagID]]);
```